core/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -167,6 +174,18 @@ protected RepositoryFileTree getTreeCore( @NonNull GetTreeOptions options ) thro
     return tree;
   }
 
+  @Override public IGenericFileContentWrapper getFileContentWrapper(@NonNull String path )
+    throws FileNotFoundException {
+    FileService fileService = getNewFileService();
+    org.pentaho.platform.api.repository2.unified.RepositoryFile repositoryFile = unifiedRepository.getFile( path );
+    RepositoryFileInputStream inputStream = fileService.getRepositoryFileInputStream( repositoryFile );
+
+    String fileName = repositoryFile.getName();
```

Comment:
if this is still valid.

I would never assume `repositoryFile` and` inputStream` are not null and throw checked exception `FileNotFoundException` unless you do want a runtime exception in this case.